main.rb

```diff
@@ -15,48 +17,40 @@ def option_list
     gets.chomp
   end
 
-  # Method that manages main option
-  # all_rentals: array of rentals
-  # all_books: array of books
-  # all_person: array of person
-  # choice: string - option value
-  # rubocop:disable Metrics/CyclomaticComplexity
-  def use_cases(choice, all_books, all_person, all_rentals)
-    case choice
-    when '1'
-      get_list_books(all_books, false)
-    when '2'
-      get_list_person(all_person, false)
-    when '3'
-      puts 'Do you want to create a student (1) or a teacher (2) [Input the number] :'
-      choice = gets.chomp
-      add_new_person(all_person, choice)
-    when '4'
-      add_new_book(all_books)
-    when '5'
-      create_new_rental(all_rentals, all_books, all_person)
-    when '6'
-      get_user_rental(all_person)
-    when '7'
-      exit
-    else
-      wrong_number_msg
-    end
+  def use_cases(choice)
+    people = gather_people
+    books = gather_books
+    rentals = gather_rentals(books, people)
+
+    data = {
+      people: people,
+      books: books,
+      rentals: rentals
+    }
+
+    options = {
+      '1' => method(:get_list_books),
+      '2' => method(:get_list_person),
+      '3' => method(:add_new_person),
+      '4' => method(:add_new_book),
+      '5' => method(:create_new_rental),
+      '6' => method(:get_user_rental),
+      '7' => method(:exit)
```

Comment:
- [ ] The exit throws an error though the data saves but it should exit gracefully without error, kindly double-check your code.